### PR TITLE
refactor: increase DataFusion batch size

### PR DESCRIPTION
**Pull request by crepererum**
See https://github.com/influxdata/idpe/issues/16900 .

### CODE REVIEW

File: datafusion_util/src/config.rs [hunk 1]
```diff
@@ -11,7 +11,7 @@ pub const DEFAULT_CATALOG: &str = "public";
 pub const DEFAULT_SCHEMA: &str = "iox";
 
 /// The maximum number of rows that DataFusion should create in each RecordBatch
-pub const BATCH_SIZE: usize = 8 * 1024;
+pub const BATCH_SIZE: usize = 32 * 1024;
```
Comment (domodwyer) [inline datafusion_util/src/config.rs:14]: What impact does this change have on the compactor? Does it increase the amount of memory datafusion allocates up front per `RecordBatch`?

Comment (aierui) [reply datafusion_util/src/config.rs:14]: I have a little problem. This value will be related to the cpu cache?

**Comment by alamb** [comment]
I do not think this is a good change as it will likely increase memory usage which is already a problem on production -- I think we should make concat batches smarter upstream. @tustvold  if we got a real reproducer, can you work on this? I can file a ticket

**Comment by tustvold** [comment]
@alamb sure, I have a couple of ideas around dictionary handling in general that I can write up. I'm not sure if the answer is making concat_batches smarter or making the coalesce smarter - need to have a think

**Comment by alamb** [comment]
> concat_batches smarter or making the coalesce smarter - need to have a think

I think coalsce batches at the least needs to know about data sizes rather than just operate on rows. I will try and write up some thoughts later

I suspect there are multiple issues -- maybe the first thing we could do is get the actual batches that are being concat'd together in the test case. Last time I dumped them via the Arrow IPC writer into local files so I could then reload them and lay around with them locally

Once we had a self contained reproducer I think we could figure out how to improve things significantly

**Comment by alamb** [comment]
I believe we have decided to fix this upstream in arrow-rs so closing this PR

**Pull request closed without merging**